paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetLoader.kt

```diff
@@ -38,64 +36,50 @@ internal interface CustomerSheetLoader {
 internal class DefaultCustomerSheetLoader(
     @Named(IS_LIVE_MODE) private val isLiveModeProvider: () -> Boolean,
     private val googlePayRepositoryFactory: @JvmSuppressWildcards (GooglePayEnvironment) -> GooglePayRepository,
-    private val elementsSessionRepository: ElementsSessionRepository,
     private val isFinancialConnectionsAvailable: IsFinancialConnectionsAvailable,
     private val lpmRepository: LpmRepository,
-    private val customerAdapterProvider: Deferred<CustomerAdapter>,
+    private val initializationDataSourceProvider: Deferred<CustomerSheetInitializationDataSource>,
     private val errorReporter: ErrorReporter,
     private val workContext: CoroutineContext
 ) : CustomerSheetLoader {
 
     @Inject constructor(
         @Named(IS_LIVE_MODE) isLiveModeProvider: () -> Boolean,
         googlePayRepositoryFactory: @JvmSuppressWildcards (GooglePayEnvironment) -> GooglePayRepository,
-        elementsSessionRepository: ElementsSessionRepository,
         isFinancialConnectionsAvailable: IsFinancialConnectionsAvailable,
         lpmRepository: LpmRepository,
         errorReporter: ErrorReporter,
         @IOContext workContext: CoroutineContext
     ) : this(
         isLiveModeProvider = isLiveModeProvider,
         googlePayRepositoryFactory = googlePayRepositoryFactory,
-        elementsSessionRepository = elementsSessionRepository,
         isFinancialConnectionsAvailable = isFinancialConnectionsAvailable,
         lpmRepository = lpmRepository,
-        customerAdapterProvider = CustomerSheetHacks.adapter,
+        initializationDataSourceProvider = CustomerSheetHacks.initializationDataSource,
         errorReporter = errorReporter,
         workContext = workContext,
     )
 
     override suspend fun load(
         configuration: CustomerSheet.Configuration
     ): Result<CustomerSheetState.Full> = workContext.runCatching {
-        val customerAdapter = retrieveCustomerAdapter().getOrThrow()
-
-        val elementsSession = retrieveElementsSession(
-            customerAdapter = customerAdapter,
-        ).getOrThrow()
+        val initializationDataSource = retrieveInitializationDataSource().getOrThrow()
```

Comment:
not for this PR, as this isn't new, but should we really be using `getOrThrow` here? I feel like in general, we want to avoid crashes. Should we have a default that we can defer to here or something?

Reply:
Or maybe we should close with a failure event here? I'm not totally sure, but I doubt we should throw here, unless we are catching all exceptions somewhere

Reply:
We are catching the exceptions with `runCatching` which is something we've been done before in `PaymentSheetLoader` and `CustomerSheetLoader`. We could look into being more explicit about our error handling. Maybe something we discuss with the rest of the team outside of this PR?

Reply:
Sounds good to me! Want to add it to our payments sdk meeting agenda? I think it'd be best practice to avoid `getOrThrow`, because I think we could accidentally call that outside of a `runCatching` or we could remove the `runCatching` without realizing the impact, etc. But we definitely don't do that now -- would be a good team discussion!

Reply:
Added to out `Payments SDK` meeting agenda.

Reply:
ty!